code/modules/xenomorph/xeno_turret.dm

```diff
@@ -168,9 +168,9 @@
 		buffer_distance = get_dist(nearby_hostile, src)
 		if(distance <= buffer_distance) //If we already found a target that's closer
 			continue
-
-		if(check_path(src, nearby_hostile, PASS_PROJECTILE) != get_turf(nearby_hostile)) //xeno turret seems to not care about actual sight, for whatever reason
+		if(check_path(get_step_towards(src, nearby_hostile), nearby_hostile, PASS_PROJECTILE) != get_turf(nearby_hostile)) //xeno turret seems to not care about actual sight, for whatever reason
```

Comment:
Jzst checking this means it wont see through walls next to it now

Reply:
yes correct, it checks the turf it starts on, which is why I needed this fix (it was blocking itself)

Reply:
Ok its working, but probs TM for a bit longer in case this effects other stuff. We really shouldn't have any situations where we want stuff working from on top of dense turfs though, so should be fine?